presto-main/src/main/java/com/facebook/presto/server/TaskInfoResource.java

```diff
@@ -69,4 +118,34 @@ public TaskInfo getTaskInfo(@PathParam("taskId") TaskId taskId)
             throw new NotFoundException(e);
         }
     }
+
+    private void proxyTaskInfo(HttpServletRequest servletRequest, AsyncResponse asyncResponse, String xForwardedProto, UriInfo uriInfo)
+    {
+        try {
+            checkState(proxyHelper.isPresent());
+            Optional<InternalNode> resourceManager = internalNodeManager.getResourceManagers().stream()
+                    .findFirst();
+            if (resourceManager.isPresent()) {
+                InternalNode resourceManagerNode = resourceManager.get();
+                URI uri = uriInfo.getRequestUriBuilder()
+                        .queryParam("includeLocalQueryOnly", true)
+                        .scheme(resourceManagerNode.getInternalUri().getScheme())
+                        .host(resourceManagerNode.getHostAndPort().toInetAddress().getHostName())
+                        .port(resourceManagerNode.getInternalUri().getPort())
+                        .build();
+                proxyHelper.get().performRequest(servletRequest, asyncResponse, uri);
+            }
+            else {
+                asyncResponse.resume(Response.status(SERVICE_UNAVAILABLE).build());
+            }
+        }
+        catch (Exception e) {
+            asyncResponse.resume(e);
+        }
+    }
+
+    private boolean requestNeedsToBeProxied(TaskId taskId, boolean includeLocalQueryOnly)
+    {
+        return resourceManagerEnabled && !dispatchManager.isQueryPresent(taskId.getQueryId()) && !includeLocalQueryOnly;
+    }
```

Comment:
keep this above proxyTaskInfo to keep the order of methods

Reply:
I would flip the order and keep checks on new line for readability.
```
return !includeLocalQueryOnly 
           && resourceManagerEnabled 
           && !dispatchManager.isQueryPresent(taskId.getQueryId());
```

Reply:
Done